diesel/src/sqlite/types/date_and_time/chrono.rs

```diff
@@ -103,12 +103,31 @@ impl ToSql<Timestamp, Sqlite> for NaiveDateTime {
     }
 }
 
+// chrono::DateTime<FixedOffset> impls
+
+impl FromSql<Text, Sqlite> for DateTime<FixedOffset> {
```

Comment:
I should have made that probably clearer in #2331, but we definitively don't want to have literally `impl FromSql<Text, Sqlite> DateTime<FixedOffset>` here, because that would:

* Remove any type safety 
* Interact poorly with the strongly typed query builder

So to clarify that a bit more, I think a better approach to handle this is to use a distinct SQL type for this. I'm not 100% sold to reusing postgres `Timestamptz` type here as I'm not sure if the semantics are similar, but that could be a solution. Alternatively we could introduce a new sql type. So the strategy should be basically similar to that one above used to implement support for `NaiveDateTime`.
Other things I've noticed while thinking about this:
* We are using [here](https://github.com/diesel-rs/diesel/pull/2351/files#diff-eff2a353ad004b64b07f479329a5076cR63-R75) some guessing for the actual timestamp format. It's probably a good idea to do that here as well with all formats that support timezone information. So we can be compatible with all sqlite supported formats and not only rfc3339 style timestamps.
* We probably want to support `DateTime<TZ: Timezone>` instead of only `DateTime<FixedOffset>`, similar to the postgres variant.